packages/webapp/src/elections/api/eden-contract.ts

```diff
@@ -293,40 +288,51 @@ const ELECTION_DEFAULTS: Election = {
 };
 
 const getParticipantsOfCompletedRounds = async (myDelegation: EdenMember[]) => {
-    console.info("getParticipantsOfCompletedRounds().top");
-    const pCompletedRounds = myDelegation.map(async (member, electionRound) => {
-        // get EdenMembers in group with this member
-        const { queryKey, queryFn } = queryParticipantsInCompletedRound(
-            electionRound,
-            member
-        );
-        const edenMembers = await queryClient.fetchQuery(queryKey, queryFn);
-
-        // get MemberDatas for these EdenMembers
-        const pParticipantsMemberData = edenMembers?.participants.map(
-            async (edenMember) => {
-                const { queryKey, queryFn } = queryMemberData(
-                    edenMember.account
-                );
-                const memberData = await queryClient.fetchQuery(
-                    queryKey,
-                    queryFn
-                );
-                return memberData;
-            }
-        );
-        const participantsMemberData = await Promise.all(
-            pParticipantsMemberData!
-        );
-
-        return {
-            participants: edenMembers?.participants, // .length will be number of participants and empty if no round happened
-            participantsMemberData,
-            didReachConsensus: isValidDelegate(
-                edenMembers?.participants?.[0]?.representative
-            ),
-        };
-    });
+    console.info(
+        "getParticipantsOfCompletedRounds().myDelegation:",
+        myDelegation
+    );
+    const pCompletedRounds = myDelegation.map(
+        async (member, electionRoundIndex) => {
+            // get EdenMembers in group with this member
+            const { queryKey, queryFn } = queryParticipantsInCompletedRound(
+                electionRoundIndex,
+                member
+            );
+            const edenMembers = await queryClient.fetchQuery(queryKey, queryFn);
+
+            // get MemberDatas for these EdenMembers
+            const pParticipantsMemberData = edenMembers?.participants.map(
+                async (edenMember) => {
+                    const { queryKey, queryFn } = queryMemberData(
+                        edenMember.account
+                    );
+                    const memberData = await queryClient.fetchQuery(
+                        queryKey,
+                        queryFn
+                    );
+                    return memberData;
+                }
+            );
+            const participantsMemberData = await Promise.all(
+                pParticipantsMemberData!
+            );
```

Comment:
We have a mechanism for querying _all_ of this `memberData` in one network call (instead of kicking off several network calls as you are here.) Modifying this to use that mechanism will save on bandwidth and RPC calls. Take a look at how we leverage `queryMembers` within `useMemberDataFromEdenMembers` in `queries.ts` to see how to do that. I highly recommend making this change.